R/num_equal.R

```diff
@@ -46,3 +45,15 @@ num_equal <- function(x, y, tolerance = default_tol()) {
 
   avg_diff < tolerance
 }
+
+
+can_int64 <- function(x) {
+  if (is.integer(x) || inherits(x, "int64")) {
+    return(TRUE)
+  }
+
+  in_range <- x >= -2^53 & x <= 2^53 - 1
```

Comment:
there is `bit64::lim.integer64()` might be used here (I guess there are systems with different `LLONG_MAX`/`LLONG_MIN`? probably only on 32-bit architectures so might be safely ignored)

Reply:
oh, i didn't see `53`. I was thinking `63`, NVM. Leaving comment since `lim.integer64()` is still useful to know.

Reply:
Added explanatory comment